homeassistant/components/kostal_plenticore/coordinator.py

```diff
@@ -237,14 +237,23 @@ class SettingDataUpdateCoordinator(
     """Implementation of PlenticoreUpdateCoordinator for settings data."""
 
     async def _async_update_data(self) -> Mapping[str, Mapping[str, str]]:
-        client = self._plenticore.client
+        if (client := self._plenticore.client) is None:
+            return {}
 
-        if not self._fetch or client is None:
+        fetch = defaultdict(set)
+
+        for module_id, data_ids in self._fetch.items():
+            fetch[module_id].update(data_ids)
+
+        for module_id, data_id in self.async_contexts():
+            fetch[module_id].add(data_id)
+
+        if not fetch:
             return {}
 
-        _LOGGER.debug("Fetching %s for %s", self.name, self._fetch)
+        _LOGGER.debug("Fetching %s for %s", self.name, fetch)
 
-        return await client.get_setting_values(self._fetch)
+        return await client.get_setting_values(fetch)
```

Comment:
Okay so
```py
self._fetch: dict[str, list[str]] = defaultdict(list)
```
it's currently a list, and now we make it a set, probably to maintain uniqueness with the context. Downside is that we now always iterate over the full `self._fetch`. I am wondering, would it make sense to make the default also ahve a set?